lib/appmap/trace/tracer.rb

```diff
@@ -135,7 +135,7 @@ def collect_self(tp)
         end
 
         def collect_parameters(tp)
-          tp.self.method(tp.method_id).parameters.collect do |pinfo|
+          -> { tp.self.method(tp.method_id).parameters rescue [] }.call.collect do |pinfo|
```

Comment:
A couple of questions about this:

* Which of the methods here might raise exceptions, and what would they be? Does it make sense to say that if an exception does get raised, that this implies that the method has no parameters?

* Why add this as a lambda, rather than a new method?